gix-worktree/src/diff.rs

```diff
@@ -257,21 +266,26 @@ pub fn compare_to_index<'a: 'b, 'b>(
                 return Some(Err(err.into()));
             }
         };
-        if index_entry.flags.contains(index::entry::Flags::INTENT_TO_ADD) {
+        if entry.flags.contains(index::entry::Flags::INTENT_TO_ADD) {
             return Some(Ok(Change {
                 kind: ChangeKind::Added,
-                index_entry,
+                entry,
                 worktree_path,
                 fstat: None,
             }));
         }
-        let mut change = match FileModification::from_stat(index_entry, &metadata, capabilities) {
+        let mut change = match FileModification::from_stat(entry, &metadata, capabilities) {
             Ok(change) => change,
             Err(err) => return Some(Err(err.into())),
         };
-        change.detect_racy_stat(index, index_entry);
+        change.detect_racy_stat(index, entry);
 
-        let kind = if change.changed() {
+        let conflict = match entry.stage() {
+            0 => false,
+            1 => true,
+            _ => return None,
+        };
+        let kind = if change.is_changed() {
```

Comment:
I saw that you moved this to the end of the function. This is very deliberately placed here so that  we don't check the same file multiple times (for different stages). Git also does the stage check in the very beginning for this reason. Why did you move it to other end here?

Reply:
That's a mistake, please move it back. I didn't recognize the `return None` as desirable side-effect but only saw the assignment and thought that should be closer to where it is used 🤦.